### ISSUE TITLE
Some inserted bounds don't pass `validate`

### ISSUE THREAD

**Issue by grovesNL** [Ralith/sieve-tree#9]
This will panic during `validate` (assuming `validate` is correct):

```rust
#[test]
fn should_validate() {
    let mut t = SieveTree::<1, 2, Bounds<1>>::new();
    let a = Bounds {
        min: [5.],
        max: [10.],
    };
    let b = Bounds {
        min: [20.],
        max: [25.],
    };
    t.insert(a, a);
    t.insert(b, b);
    validate(&t);
}
```

> element 1 origin [1500] outside of cell 8:[4] with bounds [1024]-[1279]

**Comment by grovesNL** [comment]
It looks like `validate` might be wrong in this case. If I understand correctly, it seems like the child coordinates are already being pushed to the stack:

https://github.com/Ralith/sieve-tree/blob/275ca6ea3f494b47ff972f3d1be125eaa423765d/src/lib.rs#L816

so when we pop `coords` (the child coordinates that were just pushed) from the stack later, `cell_coords` tries to account for the child offset a second time:

https://github.com/Ralith/sieve-tree/blob/275ca6ea3f494b47ff972f3d1be125eaa423765d/src/lib.rs#L825-L828

**Comment by Ralith** [comment]
Fixed in eedee1808b7cc6123246fa3df10762a0cf730dc9.
